solvebio/cli/data.py

```diff
@@ -505,6 +506,113 @@ def _download(full_path, local_folder_path, dry_run=False):
             file_.full_path, local_folder_path, file_.filename))
 
 
+def sync(args):
+    """
+    Given a folder or file, download all the files contained
+    within it (not recursive)
+    """
+    return _sync(args.full_path, args.local_path, dry_run=args.dry_run)
+
+
+def _get_remote_directory_recursive(full_path, local_folder_path, remote_files=[]):
+    """Queries a SolveBio Vault path and recursively finds all files.
+
+    full_path (str): SolveBio path where data will be downloaded from.
+    local_folder_path (str): Local folder path where data will be downloaded to.
+    remote_files (list): List used to store queried files in recursive search.
+    """
+    objects = Object.all(glob=full_path, limit=1000)
+    for object_ in objects:
+        # TODO: Skip dataset objects for now?
+        if object_.is_dataset:
+            continue
+        # add local download path
+        object_.local_path = os.path.join(local_folder_path, object_.filename)
+        remote_files.append(object_)
+        if object_.is_folder:
+            _get_remote_directory_recursive(
+                    full_path=object_.full_path + "/*",
+                    local_folder_path=object_.local_path,
+                    remote_files=remote_files)
+    return remote_files
+
+
+def _get_local_directory_recursive(local_folder_path):
```

Comment:
This isn't recursive yet or is it?

Reply:
Python built-in functions handle the recursive search for us!
https://docs.python.org/3/library/os.html#os.walk

I can rename this function though to make that more clear.